### DISCUSSION TITLE
Stack & StackItem component migration to v9.

### DISCUSSION THREAD

**Discussion by daric81** [microsoft/fluentui#28195] [General] [answered]
Hi, we use the v8 Stack and StackItem widely in our app. We are migrating v8 to v9 when components become stable. The stack shims provided still depend on @fluentui/react so long term thats a no go for us.  Has anyone successfully implemented a stack component alternative in their app or taken on the shim source and removed the dependency on v8?  We like the idea of using layout components so that all devs implement basic layout consistently.

**Comment by blendsdk** [comment]
I have the same problem. I have created a useStyle providing flex-box css and replaced the Stack with divs. It is ugly but gets the job done. I liked the Stack component. I think I am going to port my custom styles to a FlexBox (ish) component.

**Comment by mark-wiemer** [comment, marked as answer]
@spmonahan, a maintainer of this repo, recommends [CSS grid](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Grid_Layout) and [Flexbox](https://developer.mozilla.org/en-US/docs/Learn/CSS/CSS_layout/Flexbox). (https://github.com/microsoft/fluentui/discussions/23793). @micahgodbolt, another maintainer, clarifies that there will be no layout components in v9 by design (https://github.com/microsoft/fluentui/discussions/24450)

That said, there are docs for how to recreate Stack and Stack.Item on the v9 site: [@fluentui/react - Stack](https://react.fluentui.dev/?path=/docs/concepts-migration-from-v8-components-flex-stack--page).

As I'm facing the same problem, I'll probably be using this as a base implementation for a custom `Stack` component like @blendsdk.

**Reply by spmonahan** [reply]
@mark-wiemer Consider making a contrib package in [fluentui-contrib](https://github.com/microsoft/fluentui-contrib/tree/main). [This is the contributing guide](https://github.com/microsoft/fluentui-contrib/blob/main/CONTRIBUTING.md).

**Reply by mwiemer-microsoft** [reply]
Commenting on my work profile to reference this later :)

**Reply by mwiemer-microsoft** [reply]
It's been a year, and there is now a StackShim. I didn't make it though! https://react.fluentui.dev/?path=/docs/migration-shims-v8-stackshim--playground
